test/test_deconvolutional_layer.h

```diff
@@ -90,5 +90,97 @@ TEST(deconvolutional, fprop) {
     }
 }
 
+TEST(deconvolutional, gradient_check) { // tanh - mse
+    network<sequential> nn;
+    nn << deconvolutional_layer<tan_h>(2, 2, 3, 1, 1);
+
+    vec_t a(4, 0.0);
+    label_t t = 3;
+
+    uniform_rand(a.begin(), a.end(), -1, 1);
+    nn.init_weight();
+    EXPECT_TRUE(nn.gradient_check<mse>(&a, &t, 1, 1e-4, GRAD_CHECK_ALL));
```

Comment:
OK, I am rebasing now, these codes passed test as deconvolutional:[ PASSED ] in the older version.